sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -683,13 +1106,42 @@ pi_result piextQueueCreateWithNativeHandle(pi_native_handle, pi_context,
 pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             void *HostPtr, pi_mem *RetMem,
                             const pi_mem_properties *properties) {
-  (void)Context;
-  (void)Flags;
-  (void)Size;
-  (void)HostPtr;
-  (void)RetMem;
-  (void)properties;
-  DIE_NO_IMPLEMENTATION;
+  assert((Flags & PI_MEM_FLAGS_ACCESS_RW) != 0);
```

Comment:
all temporary limitations should be RT exceptions (throw sycl::exception...) rather than asserts. asserts are not triggered in release builds, users will silently get wrong results.

Reply:
Will change it.